src/components/activity/useActivityToast.tsx

```diff
@@ -0,0 +1,61 @@
+import { useWagmi } from 'libs/wagmi';
+import { useActivityQuery } from './useActivityQuery';
+import { useEffect, useState } from 'react';
+import { useStore } from 'store';
+import { TokensOverlap } from 'components/common/tokensOverlap';
+import { ReactComponent as IconClose } from 'assets/icons/X.svg';
+import { Link } from '@tanstack/react-router';
+import { lsService } from 'services/localeStorage';
+import { toPairSlug } from 'utils/pairSearch';
+
+export const useActivityToast = () => {
+  const { user } = useWagmi();
+  const [previous, setPrevious] = useState<number | null>(null);
+  const query = useActivityQuery({});
+  const allActivities = query.data || [];
+  const activities = allActivities.filter((a) => {
+    if (a.action !== 'buy' && a.action !== 'sell') return false;
+    if (user && a.strategy.owner === user) return false;
+    return true;
+  });
+  const { toaster } = useStore();
+
+  useEffect(() => {
+    if (query.fetchStatus !== 'idle') return;
+    const length = activities.length;
+    setPrevious(length);
+    if (typeof previous !== 'number' || length <= previous) return;
+    const max = Math.min(length - previous, 8);
+    for (let i = 0; i < max; i++) {
+      setTimeout(() => {
+        const preferences = lsService.getItem('notificationPreferences');
+        if (preferences?.global === false) return;
+        const { base, quote } = activities[i].strategy;
+        const id = toaster.addToast(
+          <div className="flex">
```

Comment:
I think this design has some issues. It mixes react components into hooks, it "micro manages" the toaster instead of encouraging reuse and declarative approach. For instance, the logic of getting the message id just to pass it to the callback of the close button - this is a patter that can be encapsulated in a reusable component.